lib/iris/fileformats/pp.py

```diff
@@ -1955,23 +1955,33 @@ def _convert_constraints(constraints):
     constraints = iris._constraints.list_of_constraints(constraints)
     pp_constraints = {}
     unhandled_constraints = False
+
+    def make_func(stashobj):
```

Comment:
Unfortunately, due to a technicality with lexical closures of lambda functions - it's necessary to create a new name-space for each lambda function (i.e. inside a call to `make_func`) otherwise all the lambda functions would reference the final assignment of `call_func` in the scope of the `for con in constraints:` loop (This made for quite a confusing bug...). This is because lambda functions don't copy variables, they just maintain a reference to an object until the lambda function is called.  The main suggestion online seemed to be to take this approach but perhaps there are alternatives.

Would a brief explanation of this be what you were looking for in a doc string for the function?

I agree it should be `_make_func`. :smile: